src/NuGet.Core/NuGet.ProjectModel/LockFile/AssetsLogMessage.cs

```diff
@@ -3,27 +3,43 @@
 
 using System;
 using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
 using NuGet.Common;
 using NuGet.Shared;
 
 namespace NuGet.ProjectModel
 {
     public class AssetsLogMessage : IAssetsLogMessage, IEquatable<IAssetsLogMessage>
     {
-
+        [JsonConverter(typeof(ToStringConverter<LogLevel>))]
         public LogLevel Level { get; }
+
+        [JsonConverter(typeof(ToStringConverter<NuGetLogCode>))]
         public NuGetLogCode Code { get; }
         public string Message { get; }
         public string ProjectPath { get; set; }
-        public WarningLevel WarningLevel { get; set; } = WarningLevel.Severe; //setting default to Severe as 0 implies show no warnings
+
+        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
+        public WarningLevel WarningLevel { get; set; }
         public string FilePath { get; set; }
         public string LibraryId { get; set; }
         public IReadOnlyList<string> TargetGraphs { get; set; }
+
+        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
         public int StartLineNumber { get; set; }
+
+        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
         public int StartColumnNumber { get; set; }
+
+        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
         public int EndLineNumber { get; set; }
+
+        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
         public int EndColumnNumber { get; set; }
 
+        public AssetsLogMessage() { }
```

Comment:
I'm pretty sure that System.Text.Json can still call the constructor if it's internal. If this class had `#nullable enable`, then you'd see the problem with adding this constructor: it doesn't ensure that types the rest of our code expects to always be non-null, is in fact non-null. Therefore, we'd have to make `Message` a `string?`, and now everything that processes AssetsLogMessages has to handle when `Message` is null.

Technically, there's still the problem even if this constructor is internal or private, but at least we can suppress the nullable warning explaining why it happens.  Still, I'd prefer if it could be avoided. Can we somehow use `[JsonConstructor]` on the `LogLevel, NuGetLogCode, string, string` overload?  I don't have experience with trying to tell the JsonSerializer to set some values via the constructor, and different values via properties.

For much the same reasons, a good API is one you can't use wrong, so we don't want the rest of NuGet.Client being able to construct instances where message is null. If we need this constructor for JSON deserialization, then we should limit it's potential misuse.